http/lbtransport/policy.go

```diff
@@ -146,10 +146,10 @@ func tryDial(ctx context.Context, target *Target) bool {
 		dialTimeout = deadline.Sub(time.Now())
 	}
 	// Try to do quick dial to check if the target is listening.
-	conn, err := net.DialTimeout("tcp", target.DialAddr, dialTimeout)
+	_, err := net.DialTimeout("tcp", target.DialAddr, dialTimeout)
 	if err != nil {
 		return false
 	}
-	conn.Close()
+	// Don't close connection by purpose. It will be reused anyway by this request.
```

Comment:
so conn.Close is only for custom closing by us. Connection will be closed anyway if not used. By this is extremely ugly. I changed that here: https://github.com/mwitkow/kedge/pull/40 That PR will merge into this one, so it will be removed anyway.